texastribune/donations
# Add site footer to portal + refactor some data fetching

#### What's this PR do?
+ Adds a site footer, styled with the DS (see [this PR](https://github.com/texastribune/ds-toolbox/pull/25))
+ Moves the user-data fetch to the entry point so we have said data available when the app mounts
+ Creates a protected-route mixin
+ Moves in-route data fetching to the `created` hook
+ Adds a temporary log out button for testing

#### Why are we doing this? How does it help us?
The footer is per mockups.

Moving the user-data fetch to the entry point will give us lots more flexibility in the long run if there's ever a part of this app we want to be viewable by someone who's not logged in.

#### How should this be manually tested?
+ `make`
+ `make restart`
+ Go to `/user-portal` (you'll be asked to sign into Auth0 if you're not already)
+ Confirm a footer exists and that the links work
+ Confirm the footer columns stack on mobile

#### How should this change be communicated to end users?
NA.

#### Are there any smells or added technical debt to note?
Less.

#### What are the relevant tickets?
Part of larger UMP task.

#### Have you done the following, if applicable:
